src/jdb_to_nwb/convert_behavior.py

```diff
@@ -195,21 +196,42 @@ def parse_arduino_text(arduino_text: list, arduino_timestamps: list):
 
 
 def align_data_to_visits(trial_data, block_data, metadata, logger):
+    """
+    Align arduino trial and block data to ground truth visit times.
+    Ground truth is photometry if it exists, otherwise ephys. 
+    """
+
+    # Get ground truth visit times (ground truth is photometry if it exists, otherwise ephys)
+    ground_truth_visit_times = metadata.get("ground_truth_visit_times")
+    ground_truth_time_source = metadata.get("ground_truth_time_source")
 
-    # Ground truth visits are photometry if it exists, otherwise ephys
-    ground_truth_visit_times = metadata.get("photometry_visit_times", metadata.get("ephys_visit_times"))
-    
     # If we have no ground truth visits to align to, keep trial and block data as-is
     if ground_truth_visit_times is None:
+        logger.info("No photometry or ephys visits to align to, keeping trial and block timestamps as-is.")
         return trial_data, block_data
 
     logger.info("Aligning trial and block data to ground truth port visit times...")
+    logger.info(f"Using {ground_truth_time_source} time as ground truth.")
     logger.info(f"There are {len(trial_data)} trials and {len(ground_truth_visit_times)} visit times")
 
-    assert len(trial_data) == len(ground_truth_visit_times), (
-        f"There are {len(trial_data)} trials but {len(ground_truth_visit_times)} visit times!!"
-    )
+    # If we have more ground truth visit times (pulses from photometry/ephys) than trials,
+    # trim the extra pulses so we can do alignment.
+    if len(ground_truth_visit_times) > len(trial_data):
+        logger.info(f"We have more {ground_truth_time_source} pulses than arduino visits. "
+                    "Trimming pulses to match the number of visits to do time alignment.")
+        # Port visits in arduino time are the beam break start for each trial
+        arduino_visits = [trial['beam_break_start'] for trial in trial_data]
+        ground_truth_visit_times, arduino_visits = trim_sync_pulses(ground_truth_visit_times, arduino_visits, logger)
+
+    # We should never have more trials than photometry/ephys visits.
+    # If we do, error so we can figure out why this happened and handle it accordingly.
+    elif len(trial_data) > len(ground_truth_visit_times):
+        logger.critical(f"Found more trials recorded by arduino ({len(trial_data)}) "
+                        f"than {ground_truth_time_source} visits ({len(ground_truth_visit_times)})!!!")
+        logger.critical("This should never happen!!! Skipping alignment of trial/block data.")
```

Comment:
Do you mean to raise an Exception here?

Reply:
Unsure - I think I am ok with it continuing and producing the unaligned NWB so the user can check the file for clues as to what went wrong